src/compas/files/base_reader.py

```diff
@@ -18,85 +18,164 @@ class BaseReader(object):
     ----------
     location : str or pathlib object
         Path or URL to the file
-    """
+    is_url
+
+    Methods
+    -------
+    location
+    is_url
+    is_binary
+    is_valid
+        Defined in subclasses
+    download
+        Download file specified by URL to temporary storage
+    open_ascii
+        Open ascii file specified by file path
+    open_binary
+        Open binary file specified by file path
+    iter_lines
+        Yields line from ascii file
+    iter_chunks
+        Yields chunks from binary file
+    read
+        Defined in subclasses
+    check_file_signature
+        Verify file signatures for file formats with file signature in format
+        specification
 
+    """
     def __init__(self, location):
-        self.is_url = location
-        self.location = location
+        self._location = location
+        self._is_url = None
+        self._is_binary = None
 
-    @property
-    def is_url(self):
-        return self._is_url
-
-    @is_url.setter
-    def is_url(self, location):
-        """Determine if location is local or specified as an URL
-        """
-        if str(location).startswith('http'):
-            self._is_url = True
-        else:
-            self._is_url = False
 
     @property
     def location(self):
-        return self._location
+        """ Create or ensure Pathlib object for file
+
+        Parameters
+        ----------
+        location : string or Pathlib object
+            location specified either as an URL, string containing path or
+            Pathlib object
 
-    @location.setter
-    def location(self, location):
-        """Ensure that location is specified using Path object
+        Raises
+        ------
+        IOError
+            If file is not found
         """
         if self.is_url:
-            self._location = location
+            pathobj = self.download(self._location)
         else:
-            if not isinstance(location, Path):
-                pathobj = Path(location)
+            if not isinstance(self._location, Path):
+                pathobj = Path(self._location)
             else:
-                pathobj = location
+                pathobj = self._location
 
-            if pathobj.exists():
-                self._location = pathobj
-            else:
-                raise IOError('File not found.')
+        if pathobj.exists():
+            return pathobj
 
-    def read_line_or_chunk(self, mode='ascii'):
-        """Returns iterable reading file line by line
+        raise IOError('File not found.')
 
-        Should accept both local files as well as URLs, in binary or ascii format.
+    @property
+    def is_url(self):
```

Comment:
`is_url` is not a property of the base reader. as mentioned in the docstring, it is a property of the location...

Reply:
Changed to normal method.

Reply:
that is not what i meant:
```python
reader.is_url()
```
doesn't make any sense...

Reply:
better would be
```python
reader.is_location_url()
```
or 
```python
reader.location.is_url()
```

Reply:
Ok, sorry, with you now.

Do you think it would make sense to create a new class for location?